monitor/subscription.go

```diff
@@ -69,7 +69,11 @@ func NewNodeMonitor(client *opcua.Client) (*NodeMonitor, error) {
 	return m, nil
 }
 
-func newSubscription(m *NodeMonitor, notifyChanLength int, nodes ...string) (*Subscription, error) {
+func newSubscription(m *NodeMonitor, params *opcua.SubscriptionParameters, notifyChanLength int, nodes ...string) (*Subscription, error) {
```

Comment:
Was wondering what `notifyChanLength` does here until I noticed it was there before. Looks a bit strange but we're looking at this from a different angle anyway.

Reply:
It is the size of an internal channel used for buffering results that will be sent out to the receiver.

The reason that it is parameterized (though still private), is that there are two users: `NodeMonitor.Subscribe` and `NodeMonitor.ChanSubscribe`.

In the first one (callback-based), we the library take responsibility for buffering data-change messages that are sent out via callback. Hence the larger (8192) size.

In the later, we assume the client will take responsibility for draining their user-provided channel fast enough. Therefore the internal channel only needs to be big enough to handle moving messages between underlying source channel and the user-provided channel.